### PR TITLE
[mle] suppress Announce response to orphan child on same channel and PAN ID

### PR DESCRIPTION
**Pull request by abtink**
This commit updates MLE to skip sending an Announce response when receiving an Announce message from a detached/orphan device and the included channel and PAN ID in the received Announce message are the same as the current network's channel and PAN ID.

---

Related to [SPEC-1165](https://threadgroup.atlassian.net/browse/SPEC-1165).

### CODE REVIEW

File: src/core/thread/mle.cpp [hunk 1]
```diff
@@ -3712,6 +3712,9 @@ void Mle::HandleAnnounce(RxInfo &aRxInfo)
     const MeshCoP::Timestamp *localTimestamp;
     uint8_t                   channel;
     uint16_t                  panId;
+    bool                      isFromOrhpan;
```
Comment (EskoDijk) [inline src/core/thread/mle.cpp:3715]:
```suggestion
    bool                      isFromOrphan;
```

**Review by EskoDijk — CHANGES_REQUESTED** [review]
This looks good; as it avoids sending announce triggered by devices that are on the same PANID and channel.

I was wondering where the code is that will use a mAlternateChannel/PanId from a received announce to do some attach attempt(s) with? Could not find it easily.

File: src/core/thread/mle.cpp (hunk 1 above)
Comment (abtink) [reply src/core/thread/mle.cpp]: Thanks. Fixed it in new push

**Comment by abtink** [comment]
> I was wondering where the code is that will use a mAlternateChannel/PanId from a received announce to do some attach attempt(s) with? Could not find it easily.

@EskoDijk , I think you mean this part:
https://github.com/openthread/openthread/blob/830572f81680ceb29524e215487709d3e6217121/src/core/thread/mle.cpp#L3744-L3757

which uses `mAttachTimer` to add delay before processing it from timer callback:
https://github.com/openthread/openthread/blob/830572f81680ceb29524e215487709d3e6217121/src/core/thread/mle.cpp#L1472-L1474

which then calls:

https://github.com/openthread/openthread/blob/830572f81680ceb29524e215487709d3e6217121/src/core/thread/mle.cpp#L3840-L3860

**Comment by EskoDijk** [comment]
@abtink Yes, this was the code I was looking for; thanks. Looks good now with the update.

**Review by EskoDijk — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**